src/main/java/com/example/petbuddybackend/entity/user/Client.java

```diff
@@ -25,4 +28,25 @@ public class Client {
     @OneToMany(cascade = CascadeType.ALL, mappedBy = "client", fetch = FetchType.LAZY)
     @Builder.Default
     private List<Rating> ratings = new ArrayList<>();
+
+    @ManyToMany(fetch = FetchType.LAZY)
+    @JoinTable(
+            name = "following_caretakers",
+            joinColumns = @JoinColumn(name = "client_email"),
+            inverseJoinColumns = @JoinColumn(name = "caretaker_email")
+    )
+    @Builder.Default
+    private Set<Caretaker> followingCaretakers = new HashSet<>();
+
+    @PreUpdate
+    public void preUpdate() {
+        assertClientNotFollowingItself();
+    }
+
+    private void assertClientNotFollowingItself() {
+        if (followingCaretakers.stream().anyMatch(caretaker -> caretaker.getEmail().equals(email))) {
+            throw new IllegalActionException("Client cannot follow itself");
+        }
+    }
```

Comment:
Bez PrePersist ci może nie ztriggerować tego asserta

Plus to jest dobry kandydat na `@Check` więc można (a nawet myślę że warto by było) to na poziomie bazy danych sprawdzać